lib/config.js

```diff
@@ -76,7 +76,7 @@ Config.prototype.buildArgs = function () {
     enable_widevine: true,
     target_cpu: this.targetArch,
     is_official_build: this.officialBuild,
-    is_debug: this.buildConfig !== 'Release',
+    is_debug: this.debugBuild,
```

Comment:
I think we should only add this arg if it's explicitly set by command line options

Reply:
otherwise just leave the default that gets set based on `is_component_build` inside the gn configs

Reply:
I thought it is handled here
https://github.com/brave/brave-browser/blob/d52f0d060c40c322441eb5d6268f7bfffed03043/lib/config.js#L269

Reply:
```diff --git a/lib/config.js b/lib/config.js
index e8f9760..d982134 100755
--- a/lib/config.js
+++ b/lib/config.js
@@ -35,7 +35,6 @@ const Config = function () {
   this.gypTargetArch = 'x64'
   this.targetApkBase ='classic'
   this.officialBuild = true
-  this.debugBuild = JSON.parse(getNPMConfig(['brave_debug_build']) || false)
   this.braveGoogleApiKey = getNPMConfig(['brave_google_api_key']) || 'AIzaSyAQfxPJiounkhOjODEO5ZieffeBv6yft2Q'
   this.googleApiKey = getNPMConfig(['google_api_key']) || 'AIzaSyAH90V94EcZBP5oH7oc-mXQrSKgASVxER8'
   this.googleApiEndpoint = getNPMConfig(['brave_google_api_endpoint']) || 'https://www.googleapis.com/geolocation/v1/geolocate?key='
@@ -79,7 +78,6 @@ Config.prototype.buildArgs = function () {
     target_cpu: this.targetArch,
     target_apk_base: this.targetApkBase,
     is_official_build: this.officialBuild,
-    is_debug: this.buildConfig !== 'Release',
     dcheck_always_on: this.buildConfig !== 'Release',
     brave_channel: this.channel,
     google_api_key: this.googleApiKey,
@@ -113,13 +111,16 @@ Config.prototype.buildArgs = function () {
     args.skip_signing = true
   }
 
-  if (this.debugBuild && this.targetOS !== 'ios') {
+  if (this.debugBuild !== undefined &&
+      this.debugBuild !== null &&
+      this.targetOS !== 'ios') {
     if (process.platform === 'darwin') {
       args.enable_stripping = false
     }
     args.symbol_level = 2
     args.enable_profiling = true
     args.is_win_fastlink = true
+    args.is_debug: this.debugBuild
   }
 
   if (this.sccache && process.platform === 'win32') {
@@ -328,15 +329,10 @@ Config.prototype.update = function (options) {
 
   if (options.debug_build !== null && options.debug_build !== undefined) {
     this.debugBuild = JSON.parse(options.debug_build)
-  } else {
-    this.debugBuild = this.buildConfig !== 'Release'
   }
 
   if (options.official_build !== null && options.official_build !== undefined) {
     this.officialBuild = JSON.parse(options.official_build)
-    if (this.officialBuild) {
-      this.debugBuild = false
-    }
   } else {
     this.officialBuild = this.buildConfig === 'Release'
   }
```

Reply:
we want to let gn set the default

Reply:
addressed in fd5e1f759f622902fd2203a9acdadea983bd0a07